lib/player.rb

```diff
@@ -32,17 +35,33 @@ def total_score
     return total_score
   end
 
+#returns true is a player has scored a hundred or more points
   def won?
     total_score >= 100 ? (return true) : (return false)
   end
 
+#returns the maximum of the array of player_words based on score of each word
   def highest_scoring_word
     high_word = @player_words.max_by { |word| Scrabble.score(word) }
     return high_word
   end
 
+#returns the score of the returned word of highest_scoring_word method
   def highest_word_score
     return Scrabble.score(highest_scoring_word)
   end
 
+
+#draws new tiles for the player based on TileBag draw_tiles method. This method requires a parameter, which will be a TileBag instance. We then set that tilebag instance equal to a variable and call Tilebags draw_tiles method on it. The parameter for that method is the difference between 7 (the max number of tiles a player can have) and the tile_tray, or how many tiles they already have. This returns new tiles from the Tilebag method, which we then push onto the player's tile tray and flatten into one array.
+  def draw_tiles(tile_bag = TileBag.new)
+    new_player_tiles = tile_bag.draw_tiles(7-@tile_tray.length)
+    @tile_tray.push(new_player_tiles)
```

Comment:
If you are using `@tile_tray.push(new_player_tiles)` and then `@tile_tray.flatten!` I believe you can simplify that to `@tile_tray += new_player_tiles`.